gnat2goto/driver/tree_walk.adb

```diff
@@ -2632,16 +2897,28 @@ package body Tree_Walk is
 
    function Do_Procedure_Call_Statement (N : Node_Id) return Irep
    is
-      Callee : constant String := Unique_Name (Entity (Name (N)));
+      Callee : Unbounded_String;
       --  ??? use Get_Entity_Name from gnat2why to handle entries and entry
       --  families (and most likely extend it for accesses to subprograms).
 
       Proc   : constant Irep := New_Irep (I_Symbol_Expr);
       R      : constant Irep := New_Irep (I_Code_Function_Call);
-      Sym_Id : constant Symbol_Id := Intern (Callee);
+      Sym_Id : Symbol_Id;
 
    begin
-      Set_Identifier (Proc, Callee);
+      if not (Nkind (Name (N)) in N_Has_Entity)
+        and then Nkind (Name (N)) /= N_Aspect_Specification
+        and then Nkind (Name (N)) /= N_Attribute_Definition_Clause
+        and then Nkind (Name (N)) /= N_Freeze_Entity
+        and then Nkind (Name (N)) /= N_Freeze_Generic_Entity
+      then
+         Report_Unhandled_Node_Empty (N, "Do_Procedure_Call_Statement",
+                                      "Wrong nkind of name");
+         return R;
+      end if;
+      Callee := To_Unbounded_String (Unique_Name (Entity (Name (N))));
```

Comment:
The unbounded string business here could be avoided by moving the declaration of `Callee` to down here